src/common/transformations/src/transformations/control_flow/unroll_tensor_iterator.cpp

```diff
@@ -148,8 +155,17 @@ bool ngraph::pass::UnrollTensorIterator::run_on_model(const std::shared_ptr<ngra
                     const auto& input_to_res = result->get_input_source_output(0);
                     // set output name to Tensor to store it for ngraph to cnn conversion
                     NGRAPH_SUPPRESS_DEPRECATED_START
-                    input_to_res.get_tensor().set_name(
-                        op::util::create_ie_output_name(sub_graph_op->output(concat_desc->m_output_index)));
+
+                    auto target_inputs = sub_graph_op->output(concat_desc->m_output_index).get_target_inputs();
+                    bool is_connected_to_result =
+                        std::any_of(target_inputs.begin(), target_inputs.end(), [](const ov::Input<Node>& target_inp) {
+                            return ov::as_type<opset8::Result>(target_inp.get_node()) != nullptr;
+                        });
+
+                    if (is_connected_to_result || target_inputs.empty()) {
+                        input_to_res.get_tensor().set_name(
+                            op::util::create_ie_output_name(sub_graph_op->output(concat_desc->m_output_index)));
+                    }
```

Comment:
nit: I would use lambda to avoid three copies of the same code.

Reply:
done